Basic DB setup

ttung opened the pull request:
### Description
It is an empty schema with the most rudimentary schema migration support.

Depends on #7 

#### Issue
[ch65314](https://app.clubhouse.io/genepi/stories/space/65314)

### Test plan

1. `make init-local-db` and `make drop-local-db` both work.
2. Made some dummy changes to the model code and ran `ENV=local alembic revision --autogenerate -m "SOME DESCRIPTIVE MESSAGE" --rev-id $(date +%Y%M%d_%H%M%S)` to generate a migration.  Hand checked it to see that it made sense.

phoenixAja commented on Makefile:
```diff
@@ -1,6 +1,62 @@
+
+drop-local-db:
+	@$(MAKE) stop-local-db
+	@( read -p "Delete local database container? [y/N]: " sure && case "$$sure" in [yY]) true;; *) false;; esac )
```
it's throwing me off that this @ sign is red, do you know why it's doing that?

phoenixAja replied:
i looked at the same line in the covidhub Makefile and it looks the same so i think this is correct, i was jw

ttung replied:
the `@` silences the echoing of the command when it's run.  it's mostly for aesthetic reasons.  Not sure why it's highlighted red.

phoenixAja commented on src/py/covidr/cli/db.py:
```diff
@@ -0,0 +1,19 @@
+from .toplevel import cli
+
+
+@cli.group()
```
interesting, going with click instead of argparse?

ttung replied:
I go back and forth. :)

phoenixAja commented on src/py/covidr/database/modelmixins.py:
```diff
@@ -0,0 +1,20 @@
+    id = Column(Integer, primary_key=True, autoincrement=True)
+
+
+# for things that have names
+class NameMixin:
+    """Mixin that adds a required unique name string field"""
+
+    name = Column(String, unique=True, nullable=False)
+
+    def __repr__(self):
+        return self.name
```
i'm confused when we would need this over the `BaseMixin`? Like the primary_key will always be unique so why would we need this?

phoenixAja replied:
oh nevermind i get it the comment was throwing me off

phoenixAja commented on src/py/covidr/database/models.py:
```diff
@@ -0,0 +1,17 @@
+from . import modelmixins as mx
+
+meta = MetaData(
+    schema="covidr",
```
do we need to create this schema with a migration?

phoenixAja commented on src/py/covidr/database/schema.py:
```diff
@@ -0,0 +1,28 @@
+def create_tables_and_schema(interface: SqlAlchemyInterface):
+    """Creates the schema and all tables"""
+    with session_scope(interface) as session:
+        sqlalchemy_event.listen(
+            meta,
+            "before_create",
+            DDL(f"CREATE SCHEMA IF NOT EXISTS {meta.schema}"),
+        )
```
oh ok interesting, why not just create the schema in the initial migration?

ttung replied:
The reason is that I want `make init-local-db` actually produce a functional db.

phoenixAja reviewed (approved):
looks good, just had a couple questions

Commit:
Basic DB setup

### Description
It is an empty schema with the most rudimentary schema migration support.

#### Issue
[ch65314](https://app.clubhouse.io/genepi/stories/space/65314)

### Test plan

1. `make init-local-db` and `make drop-local-db` both work.
2. Made some dummy changes to the model code and ran `ENV=local alembic revision --autogenerate -m "SOME DESCRIPTIVE MESSAGE" --rev-id $(date +%Y%M%d_%H%M%S)` to generate a migration.  Hand checked it to see that it made sense.
